proxy/proxy.go

```diff
@@ -816,7 +816,8 @@ func (c *client) interceptSystemQuery(hdr *frame.Header, stmt interface{}) {
 			c.send(hdr, &message.ServerError{ErrorMessage: "Proxy unable to create new session for keyspace"})
 		} else {
 			c.keyspace = s.Keyspace
-			c.send(hdr, &message.SetKeyspaceResult{Keyspace: s.Keyspace})
+			ks := parser.IdentifierFromString(s.Keyspace)
```

Comment:
We have a version of the proxy we're using internally. This is fixed using this patch:

```diff
diff --git a/parser/parse_select.go b/parser/parse_select.go
index 1447de1..6aa2e33 100644
--- a/parser/parse_select.go
+++ b/parser/parse_select.go
@@ -69,7 +69,7 @@ func isHandledUseStmt(l *lexer) (handled bool, stmt Statement, err error) {
 	if tkIdentifier != t {
 		return false, nil, errors.New("expected identifier after 'USE' in use statement")
 	}
-	return true, &UseStatement{Keyspace: l.identifierStr()}, nil
+	return true, &UseStatement{Keyspace: l.identifier().id}, nil
 }
```

Reply:
So with the handling of double quotes it would look something like:

```diff
diff --git a/parser/parse_select.go b/parser/parse_select.go
index 1447de1..cee7815 100644
--- a/parser/parse_select.go
+++ b/parser/parse_select.go
@@ -69,7 +69,7 @@ func isHandledUseStmt(l *lexer) (handled bool, stmt Statement, err error) {
 	if tkIdentifier != t {
 		return false, nil, errors.New("expected identifier after 'USE' in use statement")
 	}
-	return true, &UseStatement{Keyspace: l.identifierStr()}, nil
+	return true, &UseStatement{Keyspace: l.identifier().ID()}, nil
 }
 ```